scrapy/utils/python.py

```diff
@@ -388,9 +388,7 @@ def extend(self, *iterables):
         self.data = chain(self.data, *iterables)
 
     def __iter__(self):
-        return self.data.__iter__()
+        return self
 
     def __next__(self):
         return next(self.data)
-
-    next = __next__
```

Comment:
this one is technically backwards incompatible

Reply:
It is. I guess I can deprecate it instead.